### PR TITLE
Sidebar frontend

### PR DESCRIPTION
**Pull request by Waleed-dar**
Just added anchor tags in navbar and completed sidebar along with responsiveness.

### CODE REVIEW

File: src/Sidebar.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import search from "./Images/search.png";
+import detail from "./Images/detail-icon.png";
+import baropener from "./Images/bar-opener.png";
+import profile from "./Images/profile logo.png";
+
+export default function Sidebar() {
+  const [sidebar, setsidebar] = useState(true);
+  const [colour, setcolour] = useState(`bg-gray-400`);
+  const [currentSelectedItem, setCurrentSelectedItem] = useState(null);
+  /* useEffect(()=>
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Please remove comment.

File: src/Sidebar.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    {
+    },[currentSelectedItem]);
+*/
+  const controlsidebar = () => {
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: use a better name for this. Camel casing should be followed too. For example, toogleCollapse

File: src/Sidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import homeicon from "./Images/home-icon.png";
+import search from "./Images/search.png";
+import detail from "./Images/detail-icon.png";
+import baropener from "./Images/bar-opener.png";
+import profile from "./Images/profile logo.png";
+
+export default function Sidebar() {
+  const [sidebar, setsidebar] = useState(true);
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: The convention for setters is to follow camelCase. So setsidebar should be setSidebar

File: src/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  const [colour, setcolour] = useState(`bg-gray-400`);
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Same here.

File: src/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import homeicon from "./Images/home-icon.png";
+import search from "./Images/search.png";
+import detail from "./Images/detail-icon.png";
+import baropener from "./Images/bar-opener.png";
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Camel case

File: src/Sidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from "react";
+import homeicon from "./Images/home-icon.png";
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: camel case

File: src/Sidebar.jsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        }`}
+      >
+        <ul className="mt-2">
+          <button className="" onClick={controlsidebar}>
+            <img className="sm:h-5 hover:bg-slate-100" src={baropener} />
+          </button>
+
+          <li
+            className={` hover:bg-slate-100 rounded-md ${
+              currentSelectedItem == "All Countries"
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Use === instead of == as we need to strictly match the data type as well.

File: src/Sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+                ? "bg-gray-100"
+                : "bg-gray-400"
+            }`}
+            onClick={() => {
+              setCurrentSelectedItem("All Countries");
+              console.log(currentSelectedItem);
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Remove console log statement

Comment (Waleed-dar) [reply src/Sidebar.jsx]: Done from my side kinldy check it!

File: src/Sidebar.jsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                }`}
+              >
+                All Countries
+              </div>
+            </button>
+          </li>
+
+          <li
+            className={`bg-gray-400  mt-2  hover:bg-slate-100 rounded-md } ${
+              CurrentSelectedItem === "CountryDetails"
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Use constant for selected Items.

File: src/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+                ? "bg-gray-100"
+                : "bg-gray-400"
+            }`}
+            onClick={() => setCurrentSelectedItem("CountryDetails")}
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Here too

File: src/Sidebar.jsx [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+                }`}
+              >
+                Country Details
+              </div>
+            </button>
+          </li>
+
+          <li
+            className={`bg-gray-400 mt-2  hover:bg-slate-100 rounded-md ${
+              CurrentSelectedItem === "FindNeighbours"
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Here too

File: src/Sidebar.jsx [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+                ? "bg-gray-100"
+                : "bg-gray-400"
+            }`}
+            onClick={() => setCurrentSelectedItem("FindNeighbours")}
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Here too

File: src/Sidebar.jsx [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+                }`}
+              >
+                FindNeighbours
+              </div>
+            </button>
+          </li>
+
+          <li
+            className={`bg-gray-400 mt-2  hover:bg-slate-100 rounded-md ${
+              CurrentSelectedItem === "PersonalInformation"
```
Comment (usamahmed97) [inline src/Sidebar.jsx]: Replace strings with constants for all the keys

**Review by usamahmed97 — CHANGES_REQUESTED** [review]
A few changes required

File: src/Sidebar.jsx (hunk 9 above)
Comment (Waleed-dar) [reply src/Sidebar.jsx]: You mean instead of giving string direct  store them in to any constant and then compare ?

**Comment by Waleed-dar** [comment]
> A few changes required

Mentioned changes have been updated kindly review it!

File: src/ConstantData.jsx [hunk 14]
```diff
@@ -0,0 +1,4 @@
+export const Option1="All Countries";
+export const Option2="CountryDetails";
+export const Option3="FindNeighbours";
+export const Option4="PersonalInformation";
```
Comment (usamahmed97) [inline src/ConstantData.jsx]: This should be like this:

export const sidebarItems = {
ALL_COUNTRIES: "All Countries",
COUNTRY_DETAILS:"countryDetails"
// and so on.
}

**Review by usamahmed97 — CHANGES_REQUESTED** [review]
A few changes required.

**Comment by Waleed-dar** [comment]
> A few changes required.

Changes made kindly review it!

**Review by usamahmed97 — APPROVED** [review]
Looks good.

**Pull request merged**